python_scripts/ensemble_hyperparameters.py

```diff
@@ -62,8 +62,15 @@
 
 # %% [markdown]
 # We can observe that in our grid-search, the largest `max_depth` together
-# with the largest `n_estimators` led to the best generalization performance.
-#
+# with the largest `n_estimators` led to the best performance in the validation
+# set. Now we will estimate the generalization performance of the best model
+# using the test set.
```

Comment:
We might make it explicit that we will use the test set that has not been used and that with the model fitted on the full training set with the best hyperparameters found earlier.